docs/source/java/substrait.rst

```diff
@@ -0,0 +1,543 @@
+    String uriNation = "file:///data/tpch_parquet/nation.parquet";
+    String uriCustomer = "file:///data/tpch_parquet/customer.parquet";
+    String substraitPlan = "{...json Substrait plan...}";
+    ScanOptions optionsNations = new ScanOptions(/*batchSize*/ 32768);
+    ScanOptions optionsCustomer = new ScanOptions(/*batchSize*/ 32768);
+    try (
+        DatasetFactory datasetFactory = new FileSystemDatasetFactory(rootAllocator(), NativeMemoryPool.getDefault(), FileFormat.PARQUET, uriNation);
+        Dataset dataset = datasetFactory.finish();
+        Scanner scanner = dataset.newScan(optionsNations);
+        ArrowReader readerNation = scanner.scanBatches();
+        DatasetFactory datasetFactoryCustomer = new FileSystemDatasetFactory(rootAllocator(), NativeMemoryPool.getDefault(), FileFormat.PARQUET, uriCustomer);
+        Dataset datasetCustomer = datasetFactoryCustomer.finish();
+        Scanner scannerCustomer = datasetCustomer.newScan(optionsCustomer);
+        ArrowReader readerCustomer = scannerCustomer.scanBatches();
+    ) {
+      Map<String, ArrowReader> mapTableToArrowReader = new HashMap<>();
+      mapTableToArrowReader.put("NATION", readerNation);
+      mapTableToArrowReader.put("CUSTOMER", readerCustomer);
+      try (ArrowReader arrowReader = new SubstraitConsumer(rootAllocator()).runQueryNamedTables(
+          substraitPlan,
+          mapTableToArrowReader
+      )){
+        while(arrowReader.loadNextBatch()){
+          System.out.println(arrowReader.getVectorSchemaRoot().contentToTSVString());
+          System.out.println(arrowReader.getVectorSchemaRoot().getRowCount());
+        }
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+
+.. code-block:: text
+
+    // Results example:
+    FieldPath(1)	FieldPath(5)	FieldPath(8)	FieldPath(6)
+    ALGERIA	Customer#000014977	10-901-414-3869	ARaV3SU4TwhxUf
+    ALGERIA	Customer#000014975	10-318-218-3381	BzV ELDsdtukkrRf5fQ
+
+Substrait Producer
+==================
+
+Let start producing Substrait plan by `Java Substrait`_ thru `Isthmus`_ by
+CLI command:
+
+.. code-block:: sql
+
+    # schema.sql
+    CREATE TABLE CUSTOMER (
+      C_CUSTKEY BIGINT NOT NULL,
+      C_NAME VARCHAR(25),
+      C_ADDRESS VARCHAR(40),
+      C_NATIONKEY BIGINT NOT NULL,
+      C_PHONE CHAR(15),
+      C_ACCTBAL DECIMAL,
+      C_MKTSEGMENT CHAR(10),
+      C_COMMENT VARCHAR(117)
+    );
+    CREATE TABLE NATION (
+      N_NATIONKEY BIGINT NOT NULL,
+      N_NAME CHAR(25),
+      N_REGIONKEY BIGINT NOT NULL,
+      N_COMMENT VARCHAR(152)
+    );
+
+.. code-block:: bash
+
+    # define Schema DDL, Query, download Isthmus and run CLI command.
+    DDL=`cat schema.sql`
+    QUERY="SELECT n.n_name, c.c_name, c.c_phone, c.c_address FROM nation n JOIN customer c ON n.n_nationkey = c.c_nationkey"
+    ./isthmus-macOS-0.6.0 "${QUERY}" --create "${DDL}"
+
+Then, the following Susbtrait Plan will be generated:
+
+.. code-block:: json
```

Comment:
I don't think it helps to show the plan here. It's very long, and again, is this meant to be a Substrait tutorial?